### PR TITLE
Divyanshu | Test  | Implementation of Kenya VAT report - Version 1

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cv1ctb7

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 1]
```diff
@@ -238,9 +247,10 @@
                                         <strong class="table-heading">{{ report.section }}</strong>
                                     </p>
                                     <div class="table-responsive">
-                                        <table mat-table [dataSource]="report.sections" [ngClass]="{'hide-last-row bordered-table': isKenyaCompany, 'table-hide-title': !isKenyaCompany}" class="
+                                         <table mat-table [dataSource]="report.sections" [ngClass]="{'hide-last-row bordered-table': isKenyaCompany, 'table-hide-title': !isKenyaCompany}" class="
                                                 table
                                                 vat-report-table
+                                                table-hide-title table
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: remove duplicate table

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 2]
```diff
@@ -270,20 +280,21 @@
...
                                                     <ng-container *ngIf="isKenyaCompany">
-                                                        {{ activeCompany.baseCurrencySymbol }}
-                                                        {{ (element?.totalAmountOfOutputTax ?? 0) | giddhCurrency }}
+                                                        {{ activeCompany?.countryV2?.currency?.code }}
+                                                        {{ element?.totalAmountOfOutputTax | giddhCurrency }}
                                                     </ng-container>
                                                 </td>
                                                 <td mat-footer-cell *matFooterCellDef class="text-right font-weight-bold">
-                                                    {{ activeCompany.baseCurrencySymbol }}
+                                                    <ng-container *ngIf="!isKenyaCompany">{{ activeCompany.baseCurrencySymbol }}</ng-container>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                    <ng-container *ngIf="!isKenyaCompany">{{ activeCompany?.baseCurrencySymbol }}</ng-container>
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 3]
```diff
@@ -140,6 +140,8 @@ export class VatReportComponent implements OnInit, OnDestroy {
     public vatReportCurrencySymbol: string = this.vatReportCurrencyList[0].additional.symbol;
     /** Holds Current Currency Map Amount Decimal currency wise for Zimbabwe report */
     public vatReportCurrencyMap: string[];
+    /** Hold True if Current branch has Tax Number */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Correct this documentation

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 4]
```diff
@@ -239,7 +242,9 @@ export class VatReportComponent implements OnInit, OnDestroy {
                 this.selectedMonth = "";
                 this.selectedYear = "";
                 this.year = null;
-                this.loadTaxDetails();
+                if(this.hasTaxNumber || this.currentOrganizationType === OrganizationType.Company){
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: format code

File: apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ export class WithHeldSettingComponent implements OnInit, OnDestroy {
     /** This will hold the value out/in to open/close setting sidebar popup */
     public asideTaxSidebarMenuState: string = 'in';
     /** Holds With Held Form control */
-    public taxPercentage: FormControl = new FormControl(6, [Validators.max(100), Validators.required]);
+    public taxPercentage: FormControl = new FormControl('', [Validators.max(100), Validators.required]);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts:28]: Correct this documentation

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 6]
```diff
@@ -141,34 +141,39 @@
                                                     </ng-container>
                                                 </th>
                                                 <td mat-cell *matCellDef="let element" class="text-right" [attr.width]="isKenyaCompany ? '12%' : '15%'">
-                                                    {{ activeCompany.baseCurrencySymbol }}
                                                     <ng-container *ngIf="!isKenyaCompany">
+                                                        {{ activeCompany.baseCurrencySymbol }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                        {{ activeCompany?.baseCurrencySymbol }}
```

File: apps/web-giddh/src/app/settings/profile/setting.profile.component.ts [hunk 7]
```diff
@@ -1267,16 +1266,21 @@ export class SettingProfileComponent implements OnInit, OnDestroy {
                     this.store.pipe(select(state => state.session.activeCompany), takeUntil(this.destroyed$)).subscribe(activeCompany => {
                         if (activeCompany) {
                             if (this.vatSupportedCountries.includes(activeCompany.countryV2?.alpha2CountryCode)) {
-                                this.taxType = this.commonLocaleData?.app_trn;
-                                this.localeData.company_address_list = this.localeData.company_trn_list;
-                                if(activeCompany.countryV2?.alpha2CountryCode === 'ZW') {
+                                if(activeCompany.countryV2?.alpha2CountryCode === 'ZW' || activeCompany.countryV2?.alpha2CountryCode === 'KE') {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/profile/setting.profile.component.ts]:
```suggestion
                                if(activeCompany?.countryV2?.alpha2CountryCode === 'ZW' || activeCompany?.countryV2?.alpha2CountryCode === 'KE') {
```

File: apps/web-giddh/src/app/settings/profile/setting.profile.component.ts [hunk 8]
```diff
@@ -1231,16 +1231,15 @@ export class SettingProfileComponent implements OnInit, OnDestroy {
 
                 this.store.pipe(select(state => state.session.activeCompany), takeUntil(this.destroyed$)).subscribe(activeCompany => {
                     if (activeCompany) {
-                        if (activeCompany.countryV2?.alpha2CountryCode === 'ZW') {
+                        if (activeCompany.countryV2?.alpha2CountryCode === 'ZW' || activeCompany.countryV2?.alpha2CountryCode === 'KE') {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/profile/setting.profile.component.ts]:
```suggestion
                        if (activeCompany?.countryV2?.alpha2CountryCode === 'ZW' || activeCompany?.countryV2?.alpha2CountryCode === 'KE') {
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts:28]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html:145]: Done

File: apps/web-giddh/src/app/settings/profile/setting.profile.component.ts (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/profile/setting.profile.component.ts]: Done

File: apps/web-giddh/src/app/settings/profile/setting.profile.component.ts (hunk 8 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/profile/setting.profile.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 9]
```diff
@@ -141,40 +141,65 @@
...
+                                                            [currencyCode]="false"
+                                                        ></amount-field>
                                                     </ng-container>
-                                                    <ng-container *ngIf="isKenyaCompany">
-                                                        {{ (element?.totalAmount ?? 0) | giddhCurrency }}
+                                                    <ng-container *ngIf="isKenyaCompany && element?.totalAmount !== undefined">
+                                                        <amount-field
+                                                            class="d-inline-block"
+                                                            [amount]="element?.totalAmount"
+                                                            [currencySymbol]="activeCompany?.countryV2?.currency?.code"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: use currency symbol here

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 10]
```diff
@@ -141,40 +141,65 @@
...
+                                                            [currencyCode]="false"
+                                                        ></amount-field>
+                                                        </ng-conatiner>
+                                                        <ng-conatiner *ngIf="isKenyaCompany && vatReport[i]?.sections[vatReport[i]?.sections?.length -
+                                                        1]?.totalAmount !== undefined">
+                                                            <amount-field
+                                                                class="d-inline-block"
+                                                                [amount]="vatReport[i]?.sections[vatReport[i]?.sections?.length -
+                                                                1]?.totalAmount"
+                                                                [currencySymbol]="activeCompany?.countryV2?.currency?.code"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: use currency symbol here

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 11]
```diff
@@ -198,35 +227,55 @@
...
                                                     </ng-container>
                                                     <ng-container *ngIf="isKenyaCompany && vatReport[i]?.sections[vatReport[i]?.sections?.length -
-                                                    2]?.section !== element?.section">
-                                                        {{ activeCompany.baseCurrencySymbol }}
-                                                        {{ (element?.totalAmountOfOutputTax ?? 0) | giddhCurrency }}
+                                                    2]?.section !== element?.section && element?.totalAmountOfOutputTax">
+                                                        <amount-field
+                                                            class="d-inline-block"
+                                                            [amount]="element?.totalAmountOfOutputTax"
+                                                            [currencySymbol]="activeCompany?.countryV2?.currency?.code"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: use currency symbol here

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 12]
```diff
@@ -198,35 +227,55 @@
...
+                                                                [currencySymbol]="activeCompany?.baseCurrencySymbol"
+                                                                [currencyCode]="false"
+                                                            ></amount-field>
+                                                        </ng-container>
+                                                        <ng-container *ngIf="isKenyaCompany">
+                                                            <amount-field
+                                                                 class="d-inline-block"
+                                                                 [amount]="vatReport[i]?.sections[vatReport[i]?.sections?.length -
+                                                                 1]?.totalAmountOfOutputTax"
+                                                                 [currencySymbol]="activeCompany?.countryV2?.currency?.code"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: use currency symbol here

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 13]
```diff
@@ -257,35 +307,46 @@
...
+                                                            [currencyCode]="false"
+                                                            ></amount-field>
                                                     </ng-container>
                                                     <ng-container *ngIf="isKenyaCompany">
-                                                        {{ activeCompany.baseCurrencySymbol }}
-                                                        {{ (element?.totalAmountOfOutputTax ?? 0) | giddhCurrency }}
+                                                        <amount-field
+                                                            class="d-inline-block"
+                                                            [amount]="element?.totalAmountOfOutputTax"
+                                                            [currencySymbol]="activeCompany?.countryV2?.currency?.code"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: use currency symbol here

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 14]
```diff
@@ -257,35 +307,46 @@
...
                                                 </td>
                                                 <td mat-footer-cell *matFooterCellDef class="text-right font-weight-bold">
-                                                    {{ activeCompany.baseCurrencySymbol }}
-                                                    {{ (vatReport[i]?.sections[vatReport[i]?.sections?.length -
-                                                        1]?.totalAmountOfOutputTax ?? 0) | giddhCurrency }}
+                                                    <amount-field
+                                                        class="d-inline-block"
+                                                        [amount]="vatReport[i]?.sections[vatReport[i]?.sections?.length -
+                                                        1]?.totalAmountOfOutputTax ?? 0"
+                                                        [currencySymbol]="isKenyaCompany ? activeCompany?.countryV2?.currency?.code : activeCompany?.baseCurrencySymbol"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: use currency symbol here

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 15]
```diff
@@ -213,6 +215,7 @@ export class VatReportComponent implements OnInit, OnDestroy {
                     if (this.currentOrganizationType === OrganizationType.Branch) {
                         currentBranchUniqueName = this.generalService.currentBranchUniqueName;
                         this.currentBranch = cloneDeep(response.find(branch => branch?.uniqueName === currentBranchUniqueName));
+                        this.hasTaxNumber = this.currentBranch?.addresses?.filter(i => i.taxNumber.length > 0)?.length > 0;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: don't use single character variable

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 15 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 14 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 13 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 12 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 11 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 10 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 9 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**